libs/nss.c

```diff
@@ -57,53 +57,80 @@ static FILE *tcb_safe_open(const char *file, const char *name)
 	return f;
 }
 
-int _nss_tcb_getspnam_r(const char *name, struct spwd *__result_buf,
-    char *__buffer, size_t __buflen, struct spwd **__result)
+enum nss_status _nss_tcb_getspnam_r(const char *name,
+    struct spwd *__result_buf, char *__buffer, size_t __buflen, int *__errnop)
 {
 	FILE *f;
 	char *file;
 	int retval, saved_errno;
+	struct spwd **result_buf_ptr = NULL;
 
-	if (asprintf(&file, TCB_FMT, name) < 0)
+	if (asprintf(&file, TCB_FMT, name) < 0) {
+		*__errnop = EAGAIN;
```

Comment:
Should maybe be `*__errnop = errno`, because `asprintf` may set perhaps `ENOMEM` and it'd be right.

Reply:
According to the specs `EAGAIN` looks correct to me, as `NSS_STATUS_TRYAGAIN` with errnop `EAGAIN` means: "One of the functions used ran temporarily out of resources or a service is currently not available."

Reply:
The documentation says that in case of `NSS_STATUS_TRYAGAIN` only `ERANGE` has a special meaning, and everything else is as good as `EAGAIN`.  The implementation agrees with the documentation, there is some special handling for `ERANGE`, but besides that all `errno` values are equal. In fact, `__nss_getent_r` returns `EAGAIN` in case of `NSS_STATUS_TRYAGAIN` regardless of `errno` value.